keras/ops/image.py

```diff
@@ -501,3 +501,157 @@ def map_coordinates(
         fill_mode,
         fill_value,
     )
+
+
+class PadToBoundingBox(Operation):
+    def __init__(
+        self,
+        offset_height,
+        offset_width,
+        target_height,
+        target_width,
+        check_dims=True,
+    ):
+        super().__init__()
+        self.offset_height = offset_height
+        self.offset_width = offset_width
+        self.target_height = target_height
+        self.target_width = target_width
+        self.check_dims = check_dims
+
+    def call(self, image):
+        return _pad_to_bounding_box(
+            image,
+            self.offset_height,
+            self.offset_width,
+            self.target_height,
+            self.target_width,
+            self.check_dims,
+        )
+
+    def compute_output_spec(self, image):
+        out_shape = (
+            image.shape[0],
+            self.target_height,
+            self.target_width,
+            image.shape[-1],
+        )
+        if len(image.shape) == 3:
+            out_shape = out_shape[1:]
+        return KerasTensor(
+            shape=out_shape,
+            dtype=image.dtype,
+        )
+
+
+def _pad_to_bounding_box(
+    image, offset_height, offset_width, target_height, target_width, check_dims
```

Comment:
Does it support tensors with unknown shapes (e.g. with TF)? Can the offsets be scalar tensors?

Reply:
With TensorFlow, it does support unknown shapes. I tried this and it worked nicely:

```py
import tensorflow as tf
tf.compat.v1.disable_eager_execution()

image = tf.compat.v1.placeholder(tf.uint8, shape=[None, None, 3])
padded_image = tf.image.pad_to_bounding_box(image, 1, 1, 100, 200)

print(padded_image.shape)  # (100, 200, 3)
```

However, I'm not sure how to add support for all the backends. Any suggestions?

Regarding tensor arguments, `tf.image.pad_to_bounding_box` doesn't have anything specific for tensor arguments, but with tensor `offset`, it works just fine. I tried this and it worked:

```py
import numpy as np
import tensorflow as tf

x = np.arange(12).reshape(2, 2, 3)
one = tf.constant(1)
tf.image.pad_to_bounding_box(x, one, one, 4, 4, True)
```

Similarly, replacing `tf.image.pad_to_bounding_box()` with `keras.ops.image.pad_to_bounding_box()` also works without errors. I also tried it with PyTorch and Jax tensors; `offset` works fine.